scalafix/rules/src/main/scala/migrate/ExplicitImplicitsRule.scala

```diff
@@ -80,7 +80,7 @@ class ExplicitImplicitsRule[G <: Global](g: G) {
 
   }
 
-  private def collectImplicit(globalTree: g.Tree, original: Tree): Option[(List[g.Symbol], Boolean)] = {
+  private def collectImplicit(globalTree: g.Tree, original: Tree): Option[(List[g.Symbol], EtaExpansion)] = {
```

Comment:
This is only a matter of taste but I don't really like the `EtaExpansion` data type. It is created in this method to be consumed in the next method `toAddImplicit`. I suggest to merge the 3 methods together:

```scala
private def collectImplicit(globalTree: g.Tree, original: Tree): Option[Patch] = {
  val collectedTree = ...
  collectedTree.collectFirst {
    case ... =>
      val listOfArgs = ...
          val isTermEta = original.parent.forall(p => p.isInstanceOf[Term.Eta])
          if (isTermEta) EtaExpansion.WontHandle
         else
            fun match {
              case _: g.Name => Patch.addRight(...)
              //  def foo2(a: Int)(b: Int)(implicit value: Int): Int = a + b
              //  def bar(f: Int => Int => Int): Int = ???
              //  need a rewrite like this bar(a => b =>  foo(a)(b)(implicit))
              // we don't wont to handle this rewrite.
              case g.Apply(g.Apply(_, _), _) if original.isInstanceOf[Name] || original.isInstanceOf[Term.Select] =>
                Patch.empty
             case g.Apply(_, args) if original.isInstanceOf[Name] || original.isInstanceOf[Term.Select] =>
               Patch.addRight(...)
             case _ => Patch.addRight(...)
      }
      
  }
}
```

Reply:
collectimplicit, it's not the last step before printing the implicit params and creating the patch. There is still some logic about how to write the implicit. It's better to have the logic of writing the patch in one function that only handle that. 
I will propose something to remove this type